### PR TITLE
QF-3715 In Subscriptions admin add a filter by plan

### PR DESCRIPTION
**Pull request by dddpt**
Screenshot from 2024-01-31 17-50-23

### CODE REVIEW

**Review by faebebin — CHANGES_REQUESTED** [review]
As you alter the model fields check `makemigrations` and add also the migration file to the PR.
`docker compose exec app python manage.py makemigrations`
Eventhough it is not really modifying the db schema django will still create a migration.

Besides this, Looks good - thanks alot!

**Comment by dddpt** [comment]
done ;-)

File: docker-app/qfieldcloud/subscription/models.py [hunk 1]
```diff
@@ -437,6 +437,7 @@ class UpdateSubscriptionKwargs(TypedDict):
 
     plan = models.ForeignKey(
         Plan,
+        verbose_name=_("Plan"),
```
Comment (suricactus) [inline docker-app/qfieldcloud/subscription/models.py:440]: Why did you add the verbose name? What it was showing without it?

Comment (dddpt) [reply docker-app/qfieldcloud/subscription/models.py:440]: I added the verbose name for UI consistency, so that all filters have a capital letter: "By Period", "By Status", "By Plan", etc...
Otherwise it shows "By plan".

Comment (suricactus) [reply docker-app/qfieldcloud/subscription/models.py:440]: less code, more fun :)
And TBH I don't see a special reason why we need the model to be capital in this context.

File: docker-app/qfieldcloud/subscription/admin.py [hunk 2]
```diff
@@ -105,10 +105,7 @@ class SubscriptionAdmin(QFieldCloudModelAdmin):
         "status",
     )
 
-    list_filter = (
-        SubscriptionPeriodFilter,
-        "status",
-    )
+    list_filter = (SubscriptionPeriodFilter, "status", "plan")
```
Comment (suricactus) [inline docker-app/qfieldcloud/subscription/admin.py:108]:
```suggestion
    list_filter = (
        SubscriptionPeriodFilter,
        "status",
        "plan",
     )
```

Comment (suricactus) [reply docker-app/qfieldcloud/subscription/admin.py:108]: Also if you have a trailing comma?

Comment (suricactus) [reply docker-app/qfieldcloud/subscription/admin.py:108]: Does not make sense a longer line to be kept in the same line, while the original tuple was in two lines. Also checked locally, black does respect the trailing commas.

Note we always prefer to use trailing commas on lists/tuples/dicts/whatever structs that have the potential to grow in time.
